packages/core/src/generator/index.ts

```diff
@@ -174,7 +174,8 @@ export class UnoGenerator {
       }
     })
 
-    const preflightPromise = (async () => {
+    await Promise.all(tokenPromises)
+    await (async () => {
```

Comment:
Why not use out of bound promise to wait for the variables from the postprocess'? This defeats the purpose of racing tokens vs preflights as preflights might often be slower especially due to use of disk or network; compared to mostly cpu-bound token generation.

Reply:
Thank for you review.
Is there any way let us know the postprocess is all finished？

Reply:
Good point, since waiting for all postprocess cannot easily be detected due to the destructure
https://github.com/unocss/unocss/blob/a8d4db6c4c7c6e1fdbb00d98f7dd282063a4cec6/packages/core/src/generator/index.ts#L200-L201

I'd propose to upgrade the existing event system to add communications from token/preflight generation. Something [like this](https://github.com/chu121su12/unocss/commit/27838248b317b8b946b7322488e6ae2198c31929).

Reply:
Agreed, this would be useful by the way